[dataquery] Adding the test plan (#5499)
aces/Loris: modules/dataquery/test/TestPlan.md

# Data Query Tool (DQT) Test Plan

> 1. Access the DQT page with a user that has permission 'View Data Query Tool', ensure it renders

Reviewer: ```suggestion
1. Ensure the DQT module loads only when with a user that has the permission 'View Data Query Tool' (`dataquery_view`)
```

> 1. Ensure the DQT module loads only for user that has the permission 'View Data Query Tool' (`dataquery_view`)

Reviewer: ```suggestion
1. Ensure the DQT module loads only for a user that has the permission 'View Data Query Tool' (`dataquery_view`)
```

Author: sorry. This time should be good. ;)

> 1. Ensure the DQT module loads only when with user that has the permission 'View Data Query Tool' (`dataquery_view`)

Reviewer 2: ```suggestion
1. Ensure the DQT module loads only for user that has the permission 'View Data Query Tool' (`dataquery_view`)
```
small syntax problem.

Author: ooops. fixed in the last commit

2. Verify that the permission 'View Data Query Tool' is required to access the DQT page
3. Go to the "Define Fields" tab:
   - select an instrument in the instrument drop down, ensure the fields are displayed
     with the option to restrict to certain visits
   - ensure that the search of fields within the instrument is working
   - ensure that clicking on the "Add All" button adds all the fields for the 
     selected instrument to the list of fields to be queried on the
     right side of the screen
   - ensure that clicking on the "Remove All" button removes all the selected 
     fields for the selected instrument to the list of fields to be queried on the
     right side of the screen
   - select a few fields from different instruments and ensure that they are all added 
     to the list of selected fields on the right side of the screen

>    - ensure that the number of visits selected reflect the number of visits that were

Reviewer 2: Maybe rephrase that a bit differently?
```suggestion
   - ensure that the number of visits added to the query reflect the number of visits that were
```

     selected for the fields
   -
4. Go to the "Define Filters" tab:
   - add a filter by selecting an instrument; ensure that drop downs with field names
     are available for that instrument; select a field; ensure that a drop down with 
     comparison operators appears next to the selected field; select a comparison; 
     when applicable, ensure that a text box appears next to the comparison operator;
     enter a value in the text box and ensure a drop down with the list of visits 
     on which to apply the filter appears next to the text box
   - add another filter by clicking on the "Add Rule" button
   - add a group filter by clicking on the "Add Group" button
   - ensure that the delete button for a given filter deletes the filter
   - ensure that the delete button for a given group of filters deletes the group of filters
5. Go to the "View Data" tab:

>    - ensure that a "We are currently working hard to load your data. Please be patient."

Reviewer 2: ```
    "We are currently working hard to load your data. Please be patient."
```
Never knew  there were some gnomes inside DQT :-)

     message is displayed after clicking on the "Run Query" button and before the 
     data is displayed
   - ensure that a table with results is being displayed after having run the query

>    - play with the "(Maximum rows per page)" drop down and ensure the table gets updated

Reviewer 2: ```suggestion
   - Change the "(Maximum rows per page)" drop down and ensure the table gets updated
```

Author: haha. Way better phrasing indeed ;)

     accordingly
   - ensure that the first column of the data table is the "Identifiers" column that
     contains the "PSCID,VisitLabel" information separated with a comma
   - ensure that the fields that were selected are present in the resulting data table
   - ensure that the filters that were added are indeed applied to the resulting data table
   - ensure the "Download Data as ZIP" is working
   - ensure that clicking on the different page numbers on the right side are working
   - change the data view from "Cross-sectional" to "Longitudinal", ensure that the 
     data is now displayed longitudinal on the X axis with column header names changed
     to "VisitLabel,InstrumentName" separated with a comma
6. Go to the "Statistical Analysis" tab:
   - ensure that a table with "Basic Statistics" (Min, Max, Standard Deviation, Mean,
     Mean Deviation, Mean Squared Error, First Quartile, Second Quartile, Third 
     Quartile) is shown and filled
   - ensure that the "Scatterplot" is working when selecting data for column X and Y

> 7. Go to the "Manage Saved Queries" tab:

Author: @driusan This is not working so I could not really elaborate on this one. Maybe you can help? Or someone else with the DQT setup?

   - ensure the current query can be saved
   - ...

> 8. Go to the "Load Saved Query" tab:

Author: @driusan This one either is not working so I could not really elaborate. Maybe you can help? Or someone else with the DQT setup?

   - ensure that the fields that were selected are present in the resulting data table
   - ensure that the filters that were added are indeed applied to the resulting data table
   - ensure the "Download Data as ZIP" is working
   - ensure that clicking on the different page numbers on the right side are working
   - change the data view from "Cross-sectional" to "Longitudinal", ensure that the 
     data is now displayed longitudinal on the X axis with column header names changed
     to "VisitLabel,InstrumentName" separated with a comma
6. Go to the "Statistical Analysis" tab:
   - ensure that a table with "Basic Statistics" (Min, Max, Standard Deviation, Mean,
     Mean Deviation, Mean Squared Error, First Quartile, Second Quartile, Third 
     Quartile) is shown and filled
   - ensure that the "Scatterplot" is working when selecting data for column X and Y
7. Go to the "Manage Saved Queries" tab:
   - ensure the 'Save Current Query' button opens to a modal window with a text box 
     to write the name of the query to be saved, a checkbox to publicly share the 
     query and 2 buttons at the bottom ('Close' and 'Save Changes') 

>      entering a name for that query and clicking on 'Save Changes' in the modal win

Reviewer 2: ```suggestion
     entering a name for that query and clicking on 'Save Changes' in the modal window
```
??
